### DISCUSSION TITLE
Xradar Integration?

### DISCUSSION THREAD

**Discussion by syedhamidali** [CyanideCN/PyCINRAD#121] [General]
Hi @CyanideCN,

As part of the [Open Radar](https://openradarscience.org/) team, we’ve been actively developing [Xradar](https://docs.openradarscience.org/projects/xradar/en/latest/) ([GitHub](https://github.com/openradar/xradar)) to consolidate radar format readers into a single, robust package. We’ve made significant progress in integrating support for most radar formats.

I was wondering if your radar data format could also be incorporated into Xradar. If your format is already covered, that’s great! If not, and you’re interested, you might consider creating a custom [backend](https://github.com/openradar/xradar/tree/main/xradar/io/backends) following the Xarray Backend API. This would not only enable compatibility with Xradar but also make cinrad’s functionalities accessible to a broader community of users.

Looking forward to hearing your thoughts!

Best,
Hamid

**Comment by CyanideCN** [comment]
> Hi @CyanideCN,
> 
> As part of the [Open Radar](https://openradarscience.org/) team, we’ve been actively developing [Xradar](https://docs.openradarscience.org/projects/xradar/en/latest/) ([GitHub](https://github.com/openradar/xradar)) to consolidate radar format readers into a single, robust package. We’ve made significant progress in integrating support for most radar formats.
> 
> I was wondering if your radar data format could also be incorporated into Xradar. If your format is already covered, that’s great! If not, and you’re interested, you might consider creating a custom [backend](https://github.com/openradar/xradar/tree/main/xradar/io/backends) following the Xarray Backend API. This would not only enable compatibility with Xradar but also make cinrad’s functionalities accessible to a broader community of users.
> 
> Looking forward to hearing your thoughts!
> 
> Best, Hamid

Hi Hamid,

I can take a look when I have time. Is there a guide for creating a custom backend?

**Reply by syedhamidali** [reply]
Hi @CyanideCN

Here is a short tutorial from Xarray https://docs.xarray.dev/en/latest/internals/how-to-add-new-backend.html
However, it would be easier for you to follow already developed backends in Xradar. You might find most of the meta data and configuration already in other modules in Xradar, you just need to import those only.
